backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs

```diff
@@ -54,16 +56,25 @@ public async Task<UserPersonalAndTeamSlotsDto> GetUserPersonalAndTeamSlotsAsync(
                 .Where(x => x.Type == SlotType.Team)
                 .GroupBy(x => x.TeamName)
                 .Select(x =>
-                new AvailabilitySlotsGroupByTeamsDto
-                {
-                    Name = x.Key,
-                    AvailabilitySlots = x.ToList()
-                })
+                    new AvailabilitySlotsGroupByTeamsDto
+                    {
+                        Name = x.Key,
+                        AvailabilitySlots = x.ToList()
+                    })
                 .ToList();
 
-            var availabilitySlotsGroupByTeamsAndUser = new UserPersonalAndTeamSlotsDto(userSlots, availabilitySlotsGroupByTeams);
+            if (teamId is null) return new UserPersonalAndTeamSlotsDto(userSlots, availabilitySlotsGroupByTeams);
+
+            var team = await _context.Teams.FirstOrDefaultAsync(team => team.Id == teamId);
+            if (team is null)
+            {
+                throw new KeyNotFoundException("Team doesn't exist");
+            }
 
-            return availabilitySlotsGroupByTeamsAndUser;
+            return new UserPersonalAndTeamSlotsDto(
+                userSlots.Where(dto => dto.TeamName == team.Name).ToList(),
+                availabilitySlotsGroupByTeams.Where(dto => dto.Name == team.Name).ToList()
+            );
```

Comment:
It would be better to filter before data loading

Reply:
Fixed